skglm/solvers/cd_solver.py

```diff
@@ -226,32 +221,50 @@ def cd_solver(
     obj_out = []
     all_feats = np.arange(n_features)
     stop_crit = np.inf  # initialize for case n_iter=0
-    w_acc, Xw_acc = np.zeros(n_features), np.zeros(n_samples)
+    w_acc, Xw_acc = np.zeros(n_features + fit_intercept), np.zeros(n_samples)
 
     is_sparse = sparse.issparse(X)
+
+    if len(w) != n_features + fit_intercept:
+        raise ValueError(
+            "The size of coefficients should be n_features + %i, expected %i, got %i" %
```

Comment:
You can use an fstring and leverage the fact that ``0 * "some string"`` gives and empty string

```suggestion
            f"The size of coefficients should be n_features {fit_intercept * '+ 1'}"
```

Reply:
actually I liked n_features + fit_intercept: if I use fit_intercept=True, I may pass a w_init of size n_features, and then the error message does not make it clear that it's because there is an intercept that it should be n_features + 1.